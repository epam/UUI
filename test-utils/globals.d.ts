declare module "jsurl";
declare module "query-string";
