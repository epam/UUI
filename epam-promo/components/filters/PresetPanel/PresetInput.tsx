import React, { useCallback, useState } from 'react';
import css from './PresetInput.scss';
import { TextInput } from '../../inputs';
import { FlexCell } from '../../layout';
import { ITablePreset } from '@epam/uui-core';

interface IPresetInputProps {
    onCancel: () => void;
    onSuccess?: (name: string) => Promise<any>;
    preset?: ITablePreset;
}

export const PresetInput = (props: IPresetInputProps) => {
    const [presetCaption, setPresetCaption] = useState(props.preset?.name || '');

    const cancelActionHandler = useCallback(() => {
        setPresetCaption('');
        props.onCancel();
    }, [props.onCancel]);

    const acceptActionHandler = useCallback(async () => {
        if (presetCaption) {
            await props.onSuccess(presetCaption);
        }
        props.onCancel();
    }, [presetCaption]);

    const newPresetOnBlurHandler = useCallback(() => {
        if (presetCaption.length) {
            return;
        }
        props.onCancel();
    }, [presetCaption.length, props.onCancel]);

    return (
        <FlexCell cx={css.presetInputCell} minWidth={180}>
            <TextInput
                cx={css.presetInput}
                onValueChange={setPresetCaption}
                value={presetCaption}
                onCancel={cancelActionHandler}
                onAccept={acceptActionHandler}
                onBlur={newPresetOnBlurHandler}
                autoFocus
            />
        </FlexCell>
    );
};
