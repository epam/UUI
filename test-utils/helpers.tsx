import React, { ComponentType, ReactElement, ReactNode } from 'react';
import { mount } from 'enzyme';
import { renderHook } from '@testing-library/react-hooks';
import { act as reactAct } from 'react-dom/test-utils';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { ContextProvider, UuiContexts } from '@epam/uui-core';

export const delay = (ms: number = 1): Promise<void> =>
    new Promise(resolve => {
        setTimeout(resolve, ms);
    });

export const testSvc = {} as UuiContexts;

export async function mountHookWithContext<TProps, TResult>(hook: (props: TProps) => TResult, initialProps?: TProps) {
    const wrapper = ({ children }: { children?: React.ReactNode }) => (
        <ContextProvider onInitCompleted={svc => Object.assign(testSvc, svc)}>{children}</ContextProvider>
    );

    const { waitForNextUpdate, rerender, ...rest } = renderHook<TProps, TResult>(hook, { wrapper, initialProps: initialProps });

    await waitForNextUpdate();

    return {
        rerender: (props: TProps) => rerender({ ...props, children: undefined }),
        waitForNextUpdate,
        ...rest,
    };
}

export const mountWithContextAsync = async (children: ReactNode, enableLegacyContext = false) => {
    const wrapper = mount(
        <ContextProvider
            onInitCompleted={svc => {
                Object.assign(testSvc, svc);
            }}
            enableLegacyContext={enableLegacyContext}
        >
            {children}
        </ContextProvider>
    );

    await reactAct(delay);
    wrapper.update();

    return wrapper;
};

export const mountWrappedComponentAsync = async <TComponent extends ComponentType<TProps>, TProps>(
    Component: TComponent,
    props: TProps,
    enableLegacyContext = false
) => {
    const renderComponent = (properties: TProps) =>
        mount(
            React.createElement(
                (p: any) => (
                    <ContextProvider
                        onInitCompleted={svc => {
                            Object.assign(testSvc, svc);
                        }}
                        enableLegacyContext={enableLegacyContext}
                    >
                        <Component {...p} />
                    </ContextProvider>
                ),
                properties
            )
        );
    const wrapper = renderComponent(props);

    await reactAct(delay);
    wrapper.update();

    return wrapper;
};

export const renderWithContextAsync = async (component: ReactElement, enableLegacyContext = false) => {
    const result = renderer.create(
        <ContextProvider
            onInitCompleted={svc => {
                Object.assign(testSvc, svc);
            }}
            enableLegacyContext={enableLegacyContext}
        >
            {component}
        </ContextProvider>
    );

    await rendererAct(delay);
    return result.toJSON();
};
//
// global = {
//     ...global,
//     delay,
//     testSvc,
//     mountWithContextAsync,
//     mountWrappedComponentAsync,
//     renderWithContextAsync,
// } as any;

// // @ts-ignore
// global.delay = delay;
// // @ts-ignore
// global.testSvc = testSvc;
// // @ts-ignore
// global.mountWithContextAsync = mountWithContextAsync;
// // @ts-ignore
// global.mountWrappedComponentAsync = mountWrappedComponentAsync;
// // @ts-ignore
// global.renderWithContextAsync = renderWithContextAsync;
