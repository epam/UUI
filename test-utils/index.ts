export * from './helpers';
export * from './mocks';
