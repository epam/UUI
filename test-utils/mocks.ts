export const windowMock = {
    matchMedia: () => ({matches: false}),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    document: {
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
    },
} as any;