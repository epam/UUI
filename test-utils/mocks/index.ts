export * from './window';
export * from './router';
