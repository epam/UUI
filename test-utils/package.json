{
	"name": "@epam/test-utils",
	"version": "4.10.2",
  "author": "EPAM",
  "license": "MIT",
  "private": true,
  "scripts": {
    "build": "node ../uui-build/scripts/buildModule"
  },
  "dependencies": {
    "@types/react-dom": "18.0.10",
    "@types/react-test-renderer": "18.0.0",
    "@epam/uui-core": "4.10.2",
    "react-test-renderer": "18.2.0",
    "@testing-library/react": "14.0.0"
	},
  "peerDependencies": {
    "react": ">=16.0.0",
    "react-dom": ">=16.0.0"
  }
}
