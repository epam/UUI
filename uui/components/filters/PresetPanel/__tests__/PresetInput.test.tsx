import React from "react";
import { renderSnapshotWithContextAsync } from "@epam/test-utils";
import { PresetInput } from "../PresetInput";

describe('PresetInput', () => {
    it('should render correctly', async () => {
        const component =  await renderSnapshotWithContextAsync(
            <PresetInput onCancel={ jest.fn() }/>,
        );
        expect(component).toMatchSnapshot();
    });
});
