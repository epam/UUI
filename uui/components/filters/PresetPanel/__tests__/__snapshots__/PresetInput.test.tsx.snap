// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresetInput should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flexCell presetInputCell uui-presets-panel-input"
    style="min-width: 180px; flex-basis: 0px;"
  >
    <div
      class="container uui-input-box uui-focus root uui-size-36 uui-control-mode-form presetInput"
    >
      <input
        class="uui-input"
        maxlength="50"
        tabindex="0"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
